trin-beacon/src/validation.rs

```diff
@@ -135,33 +139,83 @@ impl Validator<BeaconContentKey> for BeaconValidator {
-                        |err| {
-                            anyhow!(
-                                "Historical summaries with proof has invalid SSZ bytes: {:?}",
-                                err
-                            )
-                        },
-                    )?;
+                    Self::general_summaries_validation(content, key)?;
+
+                let latest_finalized_root = self
+                    .header_oracle
+                    .read()
+                    .await
+                    .get_finalized_state_root()
+                    .await?;
+
+                Self::state_summaries_validation(
+                    fork_versioned_historical_summaries,
+                    latest_finalized_root,
+                )
+                .await?;
+            }
+        }
 
-                // Check if the historical summaries with proof epoch matches the content key epoch
-                if fork_versioned_historical_summaries
-                    .historical_summaries_with_proof
-                    .epoch
-                    != key.epoch
-                {
-                    return Err(anyhow!(
+        Ok(ValidationResult::new(true))
+    }
+}
+
+impl BeaconValidator {
+    /// General validation for the historical summaries with proof content
+    fn general_summaries_validation(
+        content: &[u8],
+        key: &HistoricalSummariesWithProofKey,
+    ) -> anyhow::Result<ForkVersionedHistoricalSummariesWithProof> {
+        let fork_versioned_historical_summaries =
+            ForkVersionedHistoricalSummariesWithProof::from_ssz_bytes(content).map_err(|err| {
+                anyhow!(
+                    "Historical summaries with proof has invalid SSZ bytes: {:?}",
+                    err
+                )
+            })?;
+
+        // Check if the historical summaries with proof epoch matches the content key epoch
+        if fork_versioned_historical_summaries
+            .historical_summaries_with_proof
+            .epoch
+            != key.epoch
+        {
+            return Err(anyhow!(
                         "Historical summaries with proof epoch does not match the content key epoch: {} != {}",
                         fork_versioned_historical_summaries.historical_summaries_with_proof.epoch,
                         key.epoch
                     ));
-                }
-
-                // TODO: Validate the historical summaries with proof against current state root
-            }
         }
+        Ok(fork_versioned_historical_summaries)
+    }
 
-        Ok(ValidationResult::new(true))
+    /// Validate historical summaries against the latest finalized state root
+    async fn state_summaries_validation(
+        fork_versioned_historical_summaries: ForkVersionedHistoricalSummariesWithProof,
+        latest_finalized_root: B256,
+    ) -> anyhow::Result<()> {
+        // Verify the historical summaries against the latest finalized state root
+
```

Comment:
*nitpick* delete this newline, though I think you can just remove this docstring since it's basically the same as the docstring for the function